### PR TITLE
Feature/auth callbacks

### PR DESCRIPTION
**Pull request by BitFracture**
- Log out buttons redirect back to current page
 - Log out of Google logs out of site too, goes back to current page
 - Log in and log out callback function decorators exist
 - Display name is populated from alternative fields if missing
 - User menu now shows email, for those users who have multiple accounts with same display names
 - Fixed scope issue with user request, which correctly requests user's name information

Please test before merging: Your account name, email, and picture should be visible when you sign in.

### CODE REVIEW

**Comment by collinsc** [comment]
I did some testing and I can create theads and comments without being logged in

OMG I am an idiot I forgot I hard coded those to use an existing id while I was waiting on this request.

**Review by collinsc — APPROVED** [review]
lgtm

File: condensation-forum/templates/body.html [hunk 1]
```diff
@@ -66,14 +66,15 @@
                             <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">
                                 <img src="{{ user.picture }}" style="height: 32px; border-radius: 16px; margin-bottom: -8px; margin-top: -8px;" /> {{user.name}} <span class="caret"></span></a>
                             <ul class="dropdown-menu">
-                                <li><a href="/logout">Sign Out</a></li>
+                                <li class="dropdown-header">{{ user.email }}</li>
```
Comment (BitFracture) [inline condensation-forum/templates/body.html:69]: @collinsc This is the entry to add the email. Can you verify your template pulled correctly?

Comment (collinsc) [reply condensation-forum/templates/body.html:69]: @BitFracture I think  I forgot a fetch somewhere in there, I was on an outdated version

**Pull request merged**